apps/web/pages/[user]/[type].tsx

```diff
@@ -124,18 +127,32 @@ async function getUserPageProps(context: GetServerSidePropsContext) {
   const { user: usernames, type: slug } = paramsSchema.parse(context.params);
   const username = usernames[0];
   const { rescheduleUid, bookingUid } = context.query;
-  const { currentOrgDomain, isValidOrgDomain } = orgDomainConfig(context.req.headers.host ?? "");
+  const { currentOrgDomain, isValidOrgDomain } = orgDomainConfig(
+    context.req.headers.host ?? "",
+    context.query.orgSlug
+  );
 
   const { ssrInit } = await import("@server/lib/ssr");
   const ssr = await ssrInit(context);
   const user = await prisma.user.findFirst({
     where: {
       username,
-      organization: isValidOrgDomain
-        ? {
-            slug: currentOrgDomain,
-          }
-        : null,
+      organization:
+        isValidOrgDomain && currentOrgDomain
+          ? {
+              OR: [
+                {
+                  slug: currentOrgDomain,
+                },
+                {
+                  metadata: {
+                    path: ["requestedSlug"],
```

Comment:
Needed to make sure we also selected an unpublished organization to not end up in a 404 page